### PR TITLE
feat(core): add configuration for experimental features

### PR DESCRIPTION
**Pull request by JimmFly**
close AF-1218 AF-1219

Added configuration for experimental features

Example:
```
const blocksuiteFeatureFlags = {
  ...
  enable_expand_database_block: {
    displayName: 'Enable Expand Database Block',
    description: 'Allows expanding database blocks for better view and management.',
    feedbackType: 'discord',
    displayChannel: ['stable', 'beta', 'canary', 'internal'],
    restrictedPlatform: 'client'
  },
    enable_ai_onboarding: {
    displayName: 'AI Onboarding',
    description: 'Enables AI onboarding.',
    displayChannel: [],
    defaultState: true,
  },
  ...
}
  
```

CleanShot 2024-08-02 at 12 26 36@2x

### CODE REVIEW

**Comment by JimmFly** [comment]
* **#7699** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7699?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JimmFly and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/components/affine/setting-modal/general-setting/experimental-features/config.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+export type FeedbackType = 'discord' | 'email' | 'github';
+
+export type Flag<K extends string> = Partial<{
+  [key in K]: {
+    displayName: string;
+    displayChannel: BuildChannel[];
+    description?: string;
+    feedbackType?: FeedbackType;
+    displayInClientOrWeb?: 'client' | 'web';
```
Comment (pengx17) [inline packages/frontend/core/src/components/affine/setting-modal/general-setting/experimental-features/config.ts]: How about change it to `restrictedPlatform`? The wording seems weird to me and confusing

File: packages/frontend/core/src/components/affine/setting-modal/general-setting/experimental-features/config.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  };
+}>;
+
+// feature flag -> display name
+export const blocksuiteFeatureFlags: Flag<keyof BlockSuiteFlags> = {
+  enable_expand_database_block: {
+    displayName: 'Enable Expand Database Block',
+    description:
```
Comment (pengx17) [inline packages/frontend/core/src/components/affine/setting-modal/general-setting/experimental-features/config.ts]: no plan for i18n?

Comment (JimmFly) [reply packages/frontend/core/src/components/affine/setting-modal/general-setting/experimental-features/config.ts]: To be honest, these flags are probably only temporary and I don’t think international support is necessary.

**Review by pengx17 — APPROVED** [review]

**Review by forehalo — COMMENTED** [review]
a RFC required

**Pull request merged**